### PR TITLE
add latest version of BazelBuild.vscode-bazel

### PR DESCRIPTION
**Pull request by marcdumais-work**
License: Apache 2.0
https://github.com/bazelbuild/vscode-bazel/blob/1f44b36b9cfbe71ee46f5b5558694eb323e044d9/LICENSE

Signed-off-by: Marc Dumais <marc.dumais@ericsson.com>

### CODE REVIEW

File: extensions.json [hunk 1]
```diff
@@ -158,6 +158,12 @@
       "id": "batisteo.vscode-django",
       "repository": "https://github.com/vscode-django/vscode-django"
     },
+    {
+      "id": "BazelBuild.vscode-bazel",
+      "repository": "https://github.com/bazelbuild/vscode-bazel",
+      "version": "0.4.0",
+      "checkout": "0.4.0"
+    },
```
Comment (jankeromnes) [inline extensions.json:166]: When I run `node add-extension https://github.com/bazelbuild/vscode-bazel --checkout`, I get this instead:

```suggestion
    {
      "id": "BazelBuild.vscode-bazel",
      "download": "https://github.com/bazelbuild/vscode-bazel/releases/download/0.4.0/vscode-bazel-0.4.0.vsix",
      "version": "0.4.0"
    },
```

**Review by jankeromnes — APPROVED** [review]
Cool, many thanks for adding this extension @marcdumais-work! 🙏

Also, FYI the new recommended way to add extensions here is to simply run:

```
node add-extension REPO_URL --checkout
```

This will almost always do exactly the right thing (e.g. here, there is an official `.vsix` release availble, so the script will prefer re-publishing that directly).

**Comment by marcdumais-work** [comment]
Thanks @jankeromnes!

**Comment by jankeromnes** [comment]
@marcdumais-work Oops, auto-publishing failed with the same error as in https://github.com/open-vsx/publish-extensions/pull/285#issuecomment-776608042:

```
Downloading https://github.com/bazelbuild/vscode-bazel/releases/download/0.4.0/vscode-bazel-0.4.0.vsix
[FAIL] Could not process extension: {
  "id": "BazelBuild.vscode-bazel",
  "download": "https://github.com/bazelbuild/vscode-bazel/releases/download/0.4.0/vscode-bazel-0.4.0.vsix",
  "version": "0.4.0"
}
Error: Extension BazelBuild.vscode-bazel version 0.4.0 is already published, but is currently inactive and therefore not visible.
```

**Comment by marcdumais-work** [comment]
@spoenemann could you similarly delete the "ghost" extension preventing this one here being re-published?

**Comment by spoenemann** [comment]
Done

**Pull request merged**